app/controllers/data_access_controller.rb

```diff
@@ -50,10 +50,22 @@ def retrieve
     retrieval.user = user
     retrieval.project = project
     retrieval.save!
-
-    new_job = Retrieval.delay.process(retrieval.id, token, cmr_env, edsc_path(request.base_url + '/'), session[:access_token])
-    Rails.logger.info("A new delayed job with ID " + new_job.id.to_s + " has been created: " + params[:project]) 
-
+    
+    queue = "Default"
+    daacs = ['NSIDC', 'LPDAAC']
+    collections = JSON.parse(params[:project])['collections']
+    daacs.each do |daac|
+      collections.each do |collection|
```

Comment:
@mreese84 how common is it for people to order collections from different providers? This makes the assumption that all collections ordered are from the same provider (at least from a queue perspective) -- I think.

Reply:
@abbottry - you're precisely right.  NSIDC and LPDAAC get a special line since their collections have been having issues (causing a bottleneck, I believe), and all the rest - which are assumed to have good behavior - are all in the 'Default' line.

Other DAACs might get their own lines eventually as well.

Reply:
It's entirely possible that a user will use a project to request data from multiple collections and that those collections could be from different DAACs. How often that happens? I don't know. But I guess I've always assumed that each collection would be its own order. So in this case, if there are 3 collections (NSIDC, LP, and any other DAAC), would they each be sent to their own queue? Or would EDSC handle everything in one queue?

This is probably an edge case, but it is realistic.

Reply:
@mreese84 I experimented with that and found that a project with multiple collections is - surprisingly - treated as a single job.  

I wrote the logic under the philosophy that priority gets the priority line even if it brings friends; therefore, if the logic finds any of the collections are from one of the priority DAACs, then it gets put into that DAACs queue.

This is also true if there is the real edge case of a project with both an NSIDC and an LPDAAC collection - that case would get sent to the first priority queue encountered as it iterates, which would be NSIDC.  The order of the priority DAACs in that array matters - slightly.

Reply:
You guys are responsive, I just came here to write `actually, i dont think that matters does it?` :)

Reply:
Good catch -- the queuing logic should check all daacs in the order, and if the order includes a priority daac put the order in that daac's queue.

Reply:
@andyfekete, that is what happens unless an order includes collections from multiple priority DAACs. Since it is a single order, it can only go to one queue.

Reply:
@mreese84 - exactly, if it is a project with a collection from more than one priority daac, then the whole thing gets put into the queue of only one of those priority daacs.

This doesn't feel like a problem to me though since projects have always been a 'single job'.  It's just a - 'feature'?  Let's go with 'feature'.